src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -1732,7 +1732,9 @@ public ConnectionDetails ParseConnectionString(string connectionString)
                 ServerName = defaultBuilder.DataSource != builder.DataSource ? builder.DataSource : null,
                 TrustServerCertificate = defaultBuilder.TrustServerCertificate != builder.TrustServerCertificate ? builder.TrustServerCertificate : false,
                 TypeSystemVersion = defaultBuilder.TypeSystemVersion != builder.TypeSystemVersion ? builder.TypeSystemVersion : null,
-                UserName = defaultBuilder.UserID != builder.UserID ? builder.UserID : null,
+                // !!! ALERT - DO NOT CHANGE USER !!!
+                // SSMS 19 treats "user" as mandatory, always set it to value from connection string builder, even if it's an empty string.
+                UserName = builder.UserID,

```

Comment:
Did you verify that there aren't any other properties that were changed to return null instead of some default value?

Reply:
We did match defaults with MSSQL extension's connection options defaults (as in package.json) in PR https://github.com/microsoft/sqltoolsservice/pull/2132 as otherwise defaults don't match and we end up populating a lot of connection options in profile, even though user did not provide them as advanced options.

But other properties are not of concern, as 'user' was used by SSMS to generate connection name where it crashed.